### DISCUSSION TITLE
Why are all examples only arrays?

### DISCUSSION THREAD

**Discussion by martinszeltins** [CodeDredd/pinia-orm#1584] [Q&A] [answered]
As I was reading the documentation, I see examples of only arrays. Users, Posts, Comments etc. What if I have a `settingsStore` pinia store with values like `appName` and  `appVersion` and I want to retrieve those values from my pinia store and update them?

It makes no sense to call a repository to get all records etc. How do I use Pinia ORM when I don't have an array?

For example:

```js
import { defineStore } from 'pinia'

export const useAppStore = defineStore('app', () => {
    const name = ref('My App')
    const nameAndVersion = computed(() => `${name.value} v1.0.0`)

    const changeName = (newName: string) => {
        name.value = newName
    }

    return { name, nameAndVersion, changeName }
}, { persist: true })

```

**Comment by CodeDredd** [comment]
If you use pinia-orm for a setting store then you can use it that way
````ts
export interface SettingStore {
  data: Setting,
  nameAndVersion: string,
  name: string
}

export default class Setting extends Model{
	static entity = settings';

	static piniaOptions = {
		state: () => ({
			data: {},
			nameAndVersion : '1.0.0',
			name : '',
		}),
		persist: true,
	};
}
````

and then you can access or update it through that way:

````ts
const settings = useRepo(Setting).pinaStore<SettingStore>().$state
console.log(settings.name)

// Update
settings.name = 'Test'

````

or for simple things if you use nuxt 3 just use `useState`

**Reply by martinszeltins** [reply]
Yikes, perhaps it is better for such cases to simply use pinia and use pinia-orm only for those cases where you actually need "table rows"?

**Reply by CodeDredd** [reply, marked as answer]
yes absolutly....pinia-orm isn't best for such case. And nothing speaks against using custom pina stores and pinia-orm models. It's totally normal to use both.

**Reply by CodeDredd** [reply]
Maybe i write someday a feature to have a singleInstane. So you could use something like `useSingleStore(Setting).name` .... but somehow i think it's not neede because you can do that all easy in a simple pinia store.
